### ISSUE TITLE
GetHTMLElement

### ISSUE THREAD

**Issue by RPA-kontor** [cloudhack/zeniverserpa_issues#26]
Jeg bruger GetHTMLElement og har valgt et element/knap som jeg ønsker robotten skal klikke på. Hvis det ønskede element så ikke findes på siden længere, klikker robotten på en tilfældig knap.

**Comment by cloudhack** [comment]
Dette blev løst på kursus dagen.
Det er vigtigt man er lidt tålmodig og ikke klikker forhurtigt :-)

**Comment by RPA-kontor** [comment]
Nej, det blev ikke løst i går. Og det handler ikke om tålmodighed. 
Jeg viste dig at robotten klikker på en tilfældig knap, hvis den ønskede knap ikke findes på siden.

**Comment by cloudhack** [comment]
Du har helt ret, jeg forvekslede dette med noget andet.
Vi skal helt klart, have diskuteret dette igen, så vi finde en løsning alle kan være enig i.
Daniel tager teten på dette, i næste uge.

**Comment by DPaaske** [comment]
Kan vi lige få ekspliciteret dette bug, evt. med et par screendumps? 
Hvilken knap skal den trykke på? I hvilket system? Hvad trykker den i stedet på?

**Comment by RPA-kontor** [comment]
Det var det jeg viste jer på kursusdagen. 
Det sker i ØS, et system som Allan har arbejdet i før. 
Den trykker på en knap som sidder på samme position som den ønskede knap, men i et anden vindue.

**Comment by DPaaske** [comment]
Beklager, jeg er med nu. 
Så vidt jeg husker, så nåede vi frem til at det hele beror på den måde vi identificerer elementer. Problemet, som du selv udpeger er, at et tilsvarende element beboer det oprindelige elements plads. Og da der ikke er angivet specifikke ID'er for de eneklte knapper, er det noget nært umuligt for os at identificere dem. Det giver naturligvis anledning til nogle uhensigtmæssigheder i tilfælde som det du sidder med. Der er bare ikke rigtig nogen hensigsmæssig måde at løse det på, uden en stor investing af tid og indsats. 
Den bedste løsning, hvis ikke der kan bruges en specific URL-sti, må være at opsætte noget logik der ser på en større del af siden, for at verificere at det er den rette side der er åben og at knappen skal aktiveres.  Det er naturligvis beklageligt at i i det her tilfælde skal lave mere arbejde, for at ramme rigtigt, end i skal med UiPath, man kan også vende den om og sige at det er ØS den er gal med og at de burde følge de principper der findes, for at sikre at UiElementer kan genkendes konsistent. 
Beklager det lange svar.

**Comment by RPA-kontor** [comment]
Okay. Men vi vil stadig gerne have det med som et udviklingsønske. 
ØS er en af de programmer vi bruger mens og jeg er lidt nervøs for om vi kan gøre robotterne robuste.

**Comment by DPaaske** [comment]
Yes, det er noteret. 
For nu, vil jeg anbefale at i anskuer øvrige elementer på siden, for at validere inden tryk.
